src/app/core/basic-datatypes/entity/display-entity/display-entity.component.ts

```diff
@@ -35,29 +36,38 @@ export class DisplayEntityComponent
   constructor(
     private entityMapper: EntityMapperService,
     private router: Router,
+    private logger: LoggingService,
     private changeDetector: ChangeDetectorRef,
   ) {
     super();
   }
 
   async ngOnInit() {
     if (!this.entityToDisplay) {
-      this.entityType = this.entityType ?? this.config;
       this.entityId = this.entityId ?? this.value;
+      this.entityType = this.entityId.includes(":")
+        ? Entity.extractTypeFromId(this.entityId)
+        : this.entityType ?? this.config;
       if (!this.entityType || !this.entityId) {
         return;
       }
-      this.entityToDisplay = await this.entityMapper.load(
-        this.entityType,
-        this.entityId,
-      );
-      this.changeDetector.detectChanges();
+      try {
+        this.entityToDisplay = await this.entityMapper.load(
+          this.entityType,
+          this.entityId,
+        );
```

Comment:
if we work with prefixed entities in future, we could move the type extraction to the entity-mapper itself and allow to only pass one parameter (the prefixed id).

Reply:
We could change that now and only construct the long ID in some cases where we currently still need it (e.g. entity details).

Reply:
I wouldn't put in a full refactoring here (let's get this PR done!). but if it simplifies things we could possibly adapt entityMapper to support that and just use it here? Otherwise just an idea for later, I would say

Reply:
It currently works with the approach, so I would leave it for now.